docs/.gitignore

```diff
@@ -0,0 +1,4 @@
+# docs-specific .gitignore
+
+# `sphinx.ext.autosummary` should output to `pythonapi/` using :toctree:
+pythonapi/
```

Comment:
Please move it to the general `.gitignore`. We keep rules for all folders there: https://github.com/microsoft/LightGBM/blob/5d3a3ea47eb8693c924ffb891387e93534f92896/.gitignore#L340

Reply:
Dropped the `docs/.gitignore` in e7cf3be. Added it to `.gitignore` instead, use should be fairly clear from context.